protocol/contracts/v2/templeLineOfCredit/TempleLineOfCredit.sol

```diff
@@ -0,0 +1,968 @@
+            : _debtTokenCache.totalDebt - _repayAmount;
+
+        debtTokenData.totalDebt = _debtTokenCache.totalDebt = _newDebt;
+
+        // Update interest rates now the total debt has been updated.
+        updateInterestRates(_debtTokenCache);
+    }
+
+    /**
+     * @dev Calculate the Utilization Ratio. 
+     * It is only relevant for DAI, where there is a debt ceiling set in the cache.
+     * Numerator = The total debt across all users for this token
+     * Denominator = The max amount which TLC can borrow from the Treasury Reserves Vault
+     */
+    function utilizationRatio(
+        DebtTokenCache memory _debtTokenCache
+    ) internal pure returns (uint256) {
+        return _debtTokenCache.trvDebtCeiling == 0
+            ? 0
+            : mulDiv(_debtTokenCache.totalDebt, 1e18, _debtTokenCache.trvDebtCeiling);
+    }
+    
+    /**
+     * @dev mulDiv with an option to round the result up or down to the nearest wei
+     */
+    function mulDivRound(uint256 x, uint256 y, uint256 denominator, bool roundUp) internal pure returns (uint256 result) {
+        result = mulDiv(x, y, denominator);
+        // See OZ Math.sol for the equivalent mulDiv() with rounding.
+        if (roundUp && mulmod(x, y, denominator) > 0) {
+            result += 1;
+        }
+    }
+
+    /**
+     * @dev Calculate the latest debt for a given account & token.
+     * Derived from the prior debt checkpoint, and the interest accumulator.
+     */
+    function currentAccountDebt(
+        DebtTokenCache memory _debtTokenCache,
+        uint128 _accountDebtCheckpoint,
+        uint256 _accountInterestAccumulator,
+        bool roundUp
+    ) internal pure returns (uint128 result) {
+        return (_accountDebtCheckpoint == 0) 
+            ? 0
+            : mulDivRound(
+                _accountDebtCheckpoint, 
+                _debtTokenCache.interestAccumulator, 
+                _accountInterestAccumulator, 
+                roundUp
+            ).encodeUInt128();
+    }
+
+    /**
+     * @dev What is the max borrow liit for a given token and 
+     * amount of collateral
+     */
+    function maxBorrowLimit(
+        DebtTokenCache memory _debtTokenCache,
+        uint256 _collateral
+    ) internal pure returns (uint256) {
+        return mulDiv(
+            _collateral * _debtTokenCache.price,
+            _debtTokenCache.config.maxLtvRatio,
+            1e36
+        );
+    }
+
+    /**
+     * @dev What is the health factor, given an amount of 
+     * collateral and debt.
+     * health = (collateral value / debt value) * max LTV Limit
+     */
+    function healthFactor(
+        DebtTokenCache memory _debtTokenCache,
+        uint256 _collateral,
+        uint256 _debt
+    ) internal pure returns (uint256) {
+        return _debt == 0
+            ? type(uint256).max
+            : mulDiv(
+                _collateral * _debtTokenCache.price,
+                _debtTokenCache.config.maxLtvRatio,
+                _debt * 1e18
```

Comment:
nit: you could define 1e36 and 1e18 as constants

Reply:
Fixed in https://github.com/TempleDAO/temple/pull/814/commits/2cf3760a6d25816f1b2ea81911375af2c6f4914b